pkg/agent/manager/cache/cache.go

```diff
@@ -474,6 +486,30 @@ func (c *Cache) diffSelectors(existingEntry, newEntry *common.RegistrationEntry,
 	}
 }
 
+func (c *Cache) selectorsDifferent(existingEntry, newEntry *common.RegistrationEntry) bool {
+	if newEntry == nil || existingEntry == nil || len(existingEntry.Selectors) != len(newEntry.Selectors) {
+		return true
+	}
+
+	set := make(selectorSet)
```

Comment:
We should use allocSelectorSet here to leverage the pool

Reply:
done.